src/manifest/errors.ts

```diff
@@ -0,0 +1,160 @@
+import {
+  ManifestCustomEventSchema,
+  ManifestCustomTypeSchema,
+  ManifestDatastoreSchema,
+  ManifestFunctionSchema,
+  ManifestOAuth2ProviderSchema,
+  ManifestWorkflowSchema,
+} from "./manifest_schema.ts";
+
+export class DuplicateWorkflowError extends Error {
```

Comment:
What do you think about reducing the number of errors here down to a set of `Duplicate<PropertyName>Error`? I.e. `DuplicateCallbackIdError` and `DuplicateNameError`. Then, each of these can be passed the various Slack constructs that require unique strings for their relevant properties. That is:

- `DuplicateCallbackIdError` can apply to either workflows or functions, and the output from this error makes clear when the error applies to functions and when to workflows.
- `DuplicateNameError` can apply to datastores, events, types, etc, and again the output from this error makes clear when the error applies to which construct.

Reply:
I agree, I was on the fence about what better way to implement this, looking back this approach should reduce the amount of boiler plate cope